ratis-server-api/src/main/java/org/apache/ratis/server/metrics/RaftLogMetrics.java

```diff
@@ -23,4 +23,6 @@
 public interface RaftLogMetrics {
   /** A log entry just has been committed. */
   void onLogEntryCommitted(LogEntryHeader header);
+  /** Read statemachine timeout */
+  void onStateMachineReadTimeout();
```

Comment:
Let's rename it to onStateMachineDataReadTimeout() and add also onStateMachineDataWriteTimeout().
```
  /** Write statemachine data timeout */
  default void onStateMachineDataWriteTimeout() {
  }

  /** Read statemachine data timeout */
  default void onStateMachineDataReadTimeout() {
  }
```